types/types.go

```diff
@@ -309,3 +334,37 @@ type AccessLog struct {
 	FilePath string `json:"file,omitempty" description:"Access log file path"`
 	Format   string `json:"format,omitempty" description:"Access log format: json | common"`
 }
+
+// HasCustomHeadersDefined checks to see if any of the custom header elements have been set
+func (h *Headers) HasCustomHeadersDefined() bool {
+	if len(h.CustomResponseHeaders) == 0 &&
+		len(h.CustomRequestHeaders) == 0 {
+		return false
+	}
+	return true
+}
+
+// HasSecureHeadersDefined checks to see if any of the secure header elements have been set
+func (h *Headers) HasSecureHeadersDefined() bool {
```

Comment:
Awesome feedback. Will implement those tests! I was manually testing, and it seemed to work, but I feel you.